### ISSUE TITLE
Infinite loop when translate params is a recursive object

### ISSUE THREAD

**Issue by dceddia** [angular-translate/angular-translate#1396]
Passing a self-referential object as the translate params causes an infinite loop when using the `sanitizeParameters` sanitization strategy:

```
RangeError: Maximum call stack size exceeded
    at String.replace (native)
    at encodeEntities (http://ajax.googleapis.com/ajax/libs/angularjs/1.4.0/angular-sanitize.js:458:5)
    at Object.chars (http://ajax.googleapis.com/ajax/libs/angularjs/1.4.0/angular-sanitize.js:520:15)
    at htmlParser (http://ajax.googleapis.com/ajax/libs/angularjs/1.4.0/angular-sanitize.js:353:36)
    at http://ajax.googleapis.com/ajax/libs/angularjs/1.4.0/angular-sanitize.js:155:7
    at htmlSanitizeValue (http://cdnjs.cloudflare.com/ajax/libs/angular-translate/2.9.0/angular-translate.js:304:12)
    at mapInterpolationParameters (http://cdnjs.cloudflare.com/ajax/libs/angular-translate/2.9.0/angular-translate.js:319:14)
    at http://cdnjs.cloudflare.com/ajax/libs/angular-translate/2.9.0/angular-translate.js:312:31
    at Object.forEach (http://ajax.googleapis.com/ajax/libs/angularjs/1.4.0/angular.js:350:20)
    at mapInterpolationParameters (http://cdnjs.cloudflare.com/ajax/libs/angular-translate/2.9.0/angular-translate.js:311:15)
```

The issue seems to be that `mapInterpolationParameters` calls itself without checking for cycles.

See the JSBin here: http://jsbin.com/cocosiwura/1/edit?html,js,output (check the console)

**Comment by knalli** [comment]
If you shoot yourself in the foot, it will probably hurt. :)

Anyway: A call stack error is not good. I've added some lines holding the state (the stack), and print out a proper message: See here in action: http://jsbin.com/vogatohote/1/edit?html,js,output

**Comment by dceddia** [comment]
Haha, well, yes :)

Would it not be better to simply ignore the recursive references? As a user I imagined that it would only sanitize the params that my translation string required (which wasn't the case, but still). I think the "principle of least surprise" would be to just sanitize everything while removing the recursive references.

See #1401

**Comment by knalli** [comment]
The error has been resolved in 2.9.2 already, but we have an open issue regarding the reaction.
